packages/itwinui-react/src/utils/functions/polymorphic.tsx

```diff
@@ -26,7 +26,13 @@ const _base = <As extends keyof JSX.IntrinsicElements = 'div'>(
       // Add tabIndex to interactive elements if not already set.
       // Workaround for Safari refusing to focus links/buttons/non-text inputs.
       if (
-        Element === 'button' ||
+        (Element === 'button' &&
+          // Button should not be disabled
+          !(
+            String(props['aria-disabled']) === 'true' ||
+            String((props as any)['data-iui-disabled']) === 'true' ||
+            (props as React.ComponentPropsWithRef<'button'>).disabled
+          )) ||
```

Comment:
The first diff is generally ok, as long as the `tabIndex` can be overridden from outside. Think of it as a default/fallback, rather than something that works everywhere out-of-the-box.

The edited diff is not ok; specifically, there shouldn't be any internal focus state or even a focused prop. The `ButtonBase` part might be ok, though I'm worried that users could override the `as` prop too easily (maybe that's fine? it's their own fault).

The correct way to handle `tabIndex` would be to do it in the place where `MenuItem` is consumed. Specifically, the place where keyboard navigation is wired up, since that's where the tabIndex really comes into play.

Also, one thing to keep in mind is that when using multi-select, only the first selected option should have `tabIndex={0}`; everything else should use `-1`. The intent of using `0` is to indicate which item gets focused initially when focus moves into the list.

Reply:
Did you try investigating why the arrow key was not working? [`a630f13`](https://github.com/iTwin/iTwinUI/pull/2021/commits/a630f13c2159d05901c8d39611b59b714ac4f164) does not look good. Specifically, you are reading a ref during render (to compute `focusableItemIndex`) and you have `onMouseEnter` spread across two different places.

For accessing the current index you might be better off using [`useListItem`](https://floating-ui.com/docs/floatinglist#uselistitem) (which requires `FloatingList`)

Reply:
> Update: After seeing the [failing unit test](https://github.com/iTwin/iTwinUI/actions/runs/9550447957/job/26322466413?pr=2021), I realized that `MenuItem` needs to be focused on hover even if it is not in a `Menu`. Thus, I brought back the `onMouseEnter` and the `.focus()` call in `MenuItem` ([7a2cad2](https://github.com/iTwin/iTwinUI/commit/7a2cad2881500e23bbad5a8ab3d39bf277ca63cb)).

This undoes the "fix" from the previous comment, leaving the issue unresolved.

You should be able to remove the `onMouseEnter` from `MenuItem` and consolidate it in `popoverGetItemProps.onMouseEnter` (using `event.currentTarget.focus()`)

Reply:
> Update: After seeing the [failing unit test]

When looking at these test logs, I noticed a [warning](https://github.com/iTwin/iTwinUI/actions/runs/9550447957/job/26322466413?pr=2021#step:7:28) about nested `<button>`s that might be worth investigating. It's not related to this thread but still likely caused by this PR.

Reply:
> To confirm, are you saying the `popoverGetItemProps` from `Menu` should call `event.currentTarget.focus()` and since `MenuItem` inside a `Menu` will spread `popoverGetItemProps`, `.focus()` will be called in `MenuItem` too?

Yes that's correct. The latest changes look good!

> Just wondering, would this be considered a breaking change? (Will add a changeset depending on what we discuss)

No, it's not a problem at all. The "focus on hover" behavior only makes sense for `Menu` anyway, so this is actually more correct. Further, `MenuItem` should never be used independently of `Menu` (I believe this is something we have discussed before in #1942).

Reply:
Revisiting an earlier part of this thread:

> Regarding the `as="ButtonBase"`, I kept it as `as="button"` and with adding the `iui-button-base` class manually so that the user's `as` prop will be respected and the styles will not break. i.e. even if the user provides an `as` prop, the `iui-button-base` class will always exist.
> 
> Since I'm not using `as={ButtonBase}`, I confirmed that `MenuItem` prevents clicks when disabled. ([code](https://github.com/iTwin/iTwinUI/blob/7b2485e19692220673d771531463f6995c4c7102/packages/itwinui-react/src/core/Menu/MenuItem.tsx#L139-L144)).

I noticed two things in the current implementation:
1. We have an `onClick` handler that just guards against the `disabled` prop.
2. We are missing `type="button"` on the `<button>` element (defaults to `type="submit"`). 

For now, we should manually add `type='button'` ourselves.

In the future, we should investigate a way to use `ListItem` and `ButtonBase` together to reduce repetition.

Reply:
Let's think about `ButtonBase` later. The current code looks fine 👍